sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/CatalogTests.java

```diff
@@ -24,6 +24,15 @@ public CompilerOptions testOptions(boolean incremental, boolean optimize) {
         return result;
     }
 
+    @Test
+    public void issue3615() {
+        this.statementsFailingInCompilation("""
+                CREATE TABLE example (
+                    inserted_xid BIGINT not null,
+                    deleted_xid BIGINT not null default null -- '9223372036854775807'
+                );""", "Nullable default value assigned to non-null column 'deleted_xid'");
```

Comment:
Shouldn't it be "Null is a assigned as a default value to non-null column XXX"? Are there other nullable values that are not NULL?

Reply:
Could be a function call in the future

Reply:
Or an expression of the other columns